File: Data Structures & Algorithms/remove-element/submission-4.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int l = 0;
        int length = nums.size();
        for(int r = 0; r < length; r++) {
            if(nums[r] != val) {
                nums[l++] = nums[r];
            }
        }

        return l;
    }
};

File: Data Structures & Algorithms/majority-element/submission-4.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut cnt = 0;

        for &num in &nums {
            if cnt == 0 {
                res = num
            }
            cnt += if num == res { 1 } else { -1 };
        }

        return res
    }
}
